src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -78,6 +79,16 @@ public function resolve(Request $request, ArgumentMetadata $argument): iterable
             throw new \LogicException(sprintf('Mapping variadic argument "$%s" is not supported.', $argument->getName()));
         }
 
+        if ($attribute instanceof MapRequestPayload) {
+            if ('array' === $argument->getType()) {
+                if (!$attribute->type) {
+                    throw new NearMissValueResolverException(sprintf('Controller "%s" requires the "$%s" argument that could not be resolved. Please set the $type argument of the #[MapRequestPayload] attribute to the type of the objects in the expected array.', $this->getPrettyName($request->attributes->get('_controller')), $argument->getName()));
```

Comment:
Sorry I wasn't clear enough: the first sentence is aleady added by ArgumentResolver so that we can remove it here and remove the getPrettyName method also:
```suggestion
                    throw new NearMissValueResolverException(sprintf('Please set the $type argument of the #[MapRequestPayload] attribute to the type of the objects in the expected array.', $this->getPrettyName($request->attributes->get('_controller')), $argument->getName()));
```

Reply:
I see. Updated.